app/code/core/Mage/Directory/Block/Data.php

```diff
@@ -169,4 +170,29 @@ public function getRegionsJs()
         Varien_Profiler::stop('TEST: ' . __METHOD__);
         return $regionsJs;
     }
+
+    /**
+     * @template T of Option[]
+     * @param T $countryOptions
+     * @return array{0: array{label: string, value: Option[]}, 1: array{label: string, value: Option[]}}|T
+     */
+    private function sortCountryOptions(array $countryOptions): array
+    {
+        $topCountryCodes = $this->helper('directory')->getTopCountryCodes();
+        $headOptions = $tailOptions = [];
+
+        foreach ($countryOptions as $countryOption) {
```

Comment:
it's "better" to not run the for each when `topCountryCodes` is empty

Reply:
@luigifab better why? performance-wise this should be totally negligible..

Reply:
is it possible that $countryOptions could be null and generate a warning on the foreach?

Reply:
> is it possible that $countryOptions could be null and generate a warning on the foreach?

no

Reply:
yes, but this one + another one + another one......................